### ISSUE TITLE
allreduce doesnot use nccl

### ISSUE THREAD

**Issue by liuyongheng123** [horovod/horovod#1135]
**Environment:**
Framework: PyTorch
Framework version: 1.1.0
Horovod version: (pull the horovod docker -python2.7 version,have no idea of the Horovod version)
MPI version: 4.0.0
CUDA version: 9.0
NCCL version: 2.3.7-1+cuda9.0 (not use in the case)
Python version: 3.6.7
OS and version: ubuntu 16.04

**Checklist:**
1. Did you search issues to find if somebody asked this question before?  yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?   yes
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?  yes

**Your question:**
Please ask your question here.
Before i use Horovodrun , i export LD_LIBRARY_PATH=/usr/local/lib  in the tetminal. Then i use timeline get the running info.Find it does not use nccl allreduce. 
And then i use mpirun ,set "-x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH=/usr/local/lib -x PATH -x NCCL_IB_DISABLE=0 -x NCCL_IB_CUDA_SUPPORT=1\
   -x NCCL_SOCKET_IFNAME=ib0  \"  these para.  in the timeline , nccl allreduce isnot used ether. In the timeline , there is also MPI-allreduce.

I am sure the libnccl.so.2.4.7 and libnccl.so.2 has install in /usr/local/lib .

**Comment by liuyongheng123** [comment]
After that， i re-install nccl lib , and export the right dir .  Then execute mpirun ... ... -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH -x PATH -x NCCL_IB_DISABLE=0 -x NCCL_IB_CUDA_SUPPORT=1\ ...
the system said ,
 ":::MLPv0.5.0 ssd 1560238199.533285856 (/mlperf/ssd/train_new.py:318) train_epoch: 0
###-localtime: 2019-06-11 07:30:01,f
 0#GPU-Iteration:      0, Loss function: 22.832, Average Loss: 0.023
 1#GPU-Iteration:      0, Loss function: 23.012, Average Loss: 0.023
lyh-ThinkStation-P520:14279:14312 [0] transport/net_socket.cc:31 NCCL WARN NET/Socket : no interface found
lyh-ThinkStation-P520:14279:14312 [0] NCCL INFO init.cc:128 -> 3
lyh-ThinkStation-P520:14279:14312 [0] NCCL INFO transport/net_socket.cc:118 -> 3
lyh-ThinkStation-P520:14279:14312 [0] NCCL INFO bootstrap.cc:18 -> 3
lyh-ThinkStation-P520:14279:14312 [0] NCCL INFO bootstrap.cc:137 -> 3
lyh-ThinkStation-P520:14279:14312 [0] NCCL INFO bootstrap.cc:158 -> 3"
but then , there is nothing output.
Does the event " NCCL WARN NET/Socket : no interface found" mean that nccl runs no-well?

**Comment by alsrgv** [comment]
@liuyongheng123, a couple of questions:
1. What's the full name of the docker image?
2. Are you specifying `--network=host` to `nvidia-docker run`?

**Comment by liuyongheng123** [comment]
1 . the full name of the docker image is , 0.16.1-tf1.12.0-torch1.0.0-mxnet1.4.0-py2.7. But i install python3.6.7 and mlperf-ssd requirement env inside.
2 . yes ,both nvidia-docker run and --network=host is set

A new issue is , i use horovodrun instead of mpirun, terminal shows that ,
[1,0]<stdout>:lyh-ThinkStation-P520:14399:14428 [0] NCCL INFO NET/Socket : Using [0]enp0s31f6:192.168.116.14<0> [1]gre_sys:fe80::d8cd:20ff:fe04:2638%gre_sys<0>
[1,0]<stdout>:lyh-ThinkStation-P520:14399:14428 [0] NCCL INFO NET/Plugin : No plugin found (libnccl-net.so).
[1,0]<stdout>:
[1,0]<stdout>:lyh-ThinkStation-P520:14399:14428 [0] misc/ibvwrap.cc:63 NCCL WARN Failed to open libibverbs.so[.1]
but program goes on.
i stop the program and use timeline to check,it shows,
![image](https://user-images.githubusercontent.com/45388934/59253789-8bff6e00-8c61-11e9-96dd-b09547689be3.png)

**Comment by alsrgv** [comment]
@liuyongheng123, excellent!  You can upgrade to a later docker with Python 3.5 `horovod/horovod:0.16.3-tf1.12.0-torch1.1.0-mxnet1.4.1-py3.5` and use `horovodrun -v ...` to print out `mpirun` if you wish to run via `mpirun`.

**Comment by liuyongheng123** [comment]
i have upgrade that docker(python3.5). but in the mlperf-ssd case, it uses (** expression) for splicing some layers. So i have to upgrade the python version to 3.6 or higher in the docker.

**Comment by alsrgv** [comment]
Got it. Python 3.6 docker is on the roadmap.
